src/averageAges.js

```diff
@@ -37,7 +48,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredPeople = people.filter(({ sex, name: personName }) =>
+    sex === 'f' && (withChildren
+      ? people.some(({ mother }) => personName === mother)
+      : true));
+
+  const averageAge = Math.round(filteredPeople.reduce(
```

Comment:
You make similar calculations for men, so it would be worth taking it out into a separate function and reusing it.